### DISCUSSION TITLE
Bug authorization on websocket endpoint, help to fix

### DISCUSSION THREAD

**Discussion by elysium31** [fastapi/fastapi#9623] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.websocket("/users/me/ws")
async def read_current_user(
    websocket: WebSocket, current_user: User = Depends(get_current_user)
):
    await websocket.accept()
    await websocket.send_json(current_user.dict())
    await websocket.close()
```

### Description

While trying to authorize on websocket endpoint, fastapi fails with exception: 
`TypeError: APIKeyCookie.__call__() missing 1 required positional argument: 'request'`

Tried to fix it it here
[https://github.com/tiangolo/fastapi/pull/9591](https://github.com/tiangolo/fastapi/pull/9591)

But tests did not pass, due to annotations

```
fastapi/security/api_key.py:86: error: Incompatible default for argument "request" (default has type "None", argument has type "Request")
fastapi/security/api_key.py:87: error: Incompatible default for argument "websocket" (default has type "None", argument has type "WebSocket")

```
Already tried: 
`request: Optional[Request]`
`request: Optional[Request, Websocket]`
`request: Optional[Union[Request, WebSocket]]`

with no luck

### Operating System

Linux

### FastAPI Version

0.95.1

### Python Version

3.10.10

**Comment by Nozavi** [comment]
Unfortunately, this is a 3-year old problem https://github.com/tiangolo/fastapi/discussions/8983 and nobody's offered a working solution until now. So, if you manage to find one (or a workaround), please let us know :)

**Reply by Kludex** [reply]
For the time being, you can subclass `APIKeyBase` and create your own `APIKeyCookie`.
